src/v/kafka/server/handlers/create_topics.cc

```diff
@@ -32,14 +32,15 @@
 
 namespace kafka {
 
-static constexpr std::array<std::string_view, 7> supported_configs{
+static constexpr std::array<std::string_view, 8> supported_configs{
   {"compression.type",
    "cleanup.policy",
    "message.timestamp.type",
    "segment.bytes",
    "compaction.strategy",
    "retention.bytes",
-   "retention.ms"}};
+   "retention.ms",
+   "recovery.source"}};
```

Comment:
recovery-on-access lazy thing will be possible when redpanda will be able to consume the log segments from S3

I'm not sure about Admin API. The reason why this was implemented this was is that we need to know where to put data first. So we need the partitions and replicas to be assigned to the nodes. Then we're downloading data from S3 on that nodes. This might probably be done through Admin API but it'll still require the same stages.